### PR TITLE
PR autolabel size + warning

### PR DESCRIPTION
**Pull request by bhack**
# What does this PR do?
This PR autolabel the PR with a configurable size label and eventually add a message for very large PRs.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

### CODE REVIEW

**Comment by ianstenbit** [comment]
This LGTM. @tanzhenyu wdyt?

**Comment by ianstenbit** [comment]
@bhack do we need to add GitHub labels that this labeler can apply?

**Comment by tanzhenyu** [comment]
> This LGTM. @tanzhenyu wdyt?

LGTM!

**Review by tanzhenyu — APPROVED** [review]

**Comment by bhack** [comment]
I don't think is required but if you want to control the labels colors is better to create it.

**Comment by ianstenbit** [comment]
> I don't think is required but if you want to control the labels colors is better to create it.

I can't seem to find a label anywhere on this PR, despite the labeler saying it has run. Am I just missing it?

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  labeler:
+    runs-on: ubuntu-latest
+    name: Label the PR size
+    steps:
+      - uses: codelytv/pr-size-labeler@v1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          xs_label: 'size/xs'
```
Comment (bhack) [inline .github/workflows/labeler.yml:13]: @ianstenbit here start the label mapping

**Comment by ianstenbit** [comment]
@bhack I manually added the size/* labels to this project and re-ran the labeler, and I still don't see a size label on this PR. Am I using this wrong or is something not working correctly?

**Comment by bhack** [comment]
I'have added the explicit permissions request.  
We could put this on hold waiting for https://github.com/CodelyTV/pr-size-labeler/issues/47

File: .github/workflows/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: labeler
+
+on: [pull_request]
+
+jobs:
+  labeler:
+    runs-on: ubuntu-latest
+    name: Label the PR size
+    permissions:
```
[the remark this answers is not in the document]
Comment (ianstenbit) [reply .github/workflows/labeler.yml:9]: Looks like (according to my view of https://github.com/keras-team/keras-cv/settings/actions) we have read+write permissions by default for GITHUB_TOKEN

**Comment by bhack** [comment]
@ianstenbit Can we check again the execution?

**Comment by ianstenbit** [comment]
> @ianstenbit Can we check again the execution?

I don't see a GitHub action running for the labeler -- is it configured to run on every PR?

**Comment by bhack** [comment]
> I don't see a GitHub action running for the labeler -- is it configured to run on every PR?

Yes also I don't the the action running. I've set

```yaml
on:
  pull_request_target:
    types: [assigned, opened, synchronize, reopened]
```

**Comment by ianstenbit** [comment]
No dice on re-opening.

At https://github.com/keras-team/keras-cv/actions/workflows/labeler.yml I see runs from December but none from today.

File: .github/workflows/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: labeler
+
+on:
```
Comment (ianstenbit) [inline .github/workflows/labeler.yml:3]: Maybe just:
`on: [pull_request]`
?

**Comment by ianstenbit** [comment]
https://github.com/keras-team/keras-cv/actions/runs/4010140081/jobs/6886298137

Looks like the labeler ran, and decided to apply a label of `size/s`, but I see no such label

**Comment by bhack** [comment]
Yes it ran as one month ago with ` on: [pull_request]` but it seems it cannot write the label. It is why I've tried to change the event with the permission.
In any case the Action it is quite unmaintained at upstream level.
Can we rely on https://github.com/apps/google-ml-butler ?

**Comment by LukeWood** [comment]
> Yes it ran as one month ago with  on: [pull_request] but it seems it cannot write the label. It is why I've tried to change the event with the permission.
In any case the Action is quite unmaintained at upstream level.
Can we rely on https://github.com/apps/google-ml-butler ?

Due to these closing this out for now.  If you have an alternative feel free to reopen @bhack !  

If you do end up pursuing this, one really useful label would be "Stale" for PRs that do not have changes in 2 weeks.  That would greatly reduce our triaging efforts!

**Comment by bhack** [comment]
> One really useful label would be "Stale" for PRs that do not have changes in 2 weeks. That would greatly reduce our triaging efforts!

I've contributed this few months ago in TF and now it is regularly maintained at https://github.com/tensorflow/tensorflow/blob/master/.github/workflows/stale-issues.yml

You can just adopt that one and change the params.

**Comment by LukeWood** [comment]
Thanks for the heads up!
